### PR TITLE
Restructure library code

### PR DESCRIPTION
**Pull request by HyperGH**
Aims to restructure the library to be more modular, easier to build and maintain. The categories defined are fairly loose given that the utility methods themselves are not always that closely related, but I'm looking for feedback on improving the structure.

### CODE REVIEW

File: toolbox/members.py [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+import typing as t
+
+import hikari
+
+from .errors import CacheFailureError
+
+__all__: t.Sequence[str] = ("sort_roles", "get_member_color", "is_above", "get_possessive")
+
+
+def sort_roles(roles: t.Sequence[hikari.Role], ascending: bool = False) -> t.Sequence[hikari.Role]:
```
Comment (Lunarmagpie) [inline toolbox/members.py:12]: The functions relating to roles can be separated into a roles.py file

File: toolbox/strings.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    if style and style not in VALID_TIMESTAMP_STYLES:
+        raise ValueError(f"Invalid style passed. Valid styles: {' '.join(VALID_TIMESTAMP_STYLES)}")
+
+    if style:
+        return f"<t:{int(time.timestamp())}:{style}>"
+
+    return f"<t:{int(time.timestamp())}>"
+
+
+def utcnow() -> datetime.datetime:
```
Comment (Lunarmagpie) [inline toolbox/strings.py:43]: This definitely does not relate to strings

File: toolbox/commands.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
+import typing as t
```
Comment (Lunarmagpie) [inline toolbox/commands.py:1]: This file should be `command_choices.py`

**Comment by Lunarmagpie** [comment]
I think the string and url functions can be separated. Same with permissions and member functions.
Also the tests have to be restructured to follow the same file format.

File: toolbox/strings.py (hunk 2 above)
Comment (HyperGH) [reply toolbox/strings.py:43]: Honestly not quite sure where to put it.

File: toolbox/commands.py (hunk 3 above)
Comment (HyperGH) [reply toolbox/commands.py:1]: My idea was that other utilities relating to commands could also go here.

**Comment by HyperGH** [comment]
> I think the string and url functions can be separated. Same with permissions and member functions. Also the tests have to be restructured to follow the same file format.

That would basically mean 1function per file, which we *could* do, but I'd rather not for the moment.

File: toolbox/strings.py (hunk 2 above)
Comment (Lunarmagpie) [reply toolbox/strings.py:43]: Can do `time.py` or `utcnow.py`

**Comment by Lunarmagpie** [comment]
> That would basically mean 1 function per file, which we _could_ do, but I'd rather not for the moment.

I think it's fine to split url and string up a bit more tbh. String will get a lot added to it anyway once the markdown PR is merged.

File: toolbox/strings.py (hunk 2 above)
Comment (parafoxia) [reply toolbox/strings.py:43]: I agree, though I would recommend avoid using time.py as the filename due to potential namespace conflicts.

**Comment by Lunarmagpie** [comment]
One last thing: I think embeds.py and messages.py can be merged. (As embeds are never used outside messages).
Tests just need to be restrctured to fit the lib file structure before I give a positive review.

**Review by Lunarmagpie — CHANGES_REQUESTED** [review]
the `test_fetch_message_from_link` test should be in `tests/toolbox/test_messages.py`

**Review by Lunarmagpie — APPROVED** [review]

**Pull request merged**